src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTestHelpers.cs

```diff
@@ -35,14 +35,426 @@ public static void VerifyDisposed(SlhDsa slhDsa)
-            Assert.Throws<ObjectDisposedException>(() => slhDsa.ExportSlhDsaPublicKey(tempBuffer));
-            Assert.Throws<ObjectDisposedException>(() => slhDsa.ExportSlhDsaSecretKey(tempBuffer));
+            Assert.Throws<ObjectDisposedException>(() => slhDsa.ExportSlhDsaPublicKey(tempBuffer.AsSpan(..slhDsa.Algorithm.PublicKeySizeInBytes)));
+            Assert.Throws<ObjectDisposedException>(() => slhDsa.ExportSlhDsaSecretKey(tempBuffer.AsSpan(..slhDsa.Algorithm.SecretKeySizeInBytes)));
             Assert.Throws<ObjectDisposedException>(() => slhDsa.ExportSubjectPublicKeyInfo());
             Assert.Throws<ObjectDisposedException>(() => slhDsa.ExportSubjectPublicKeyInfoPem());
             Assert.Throws<ObjectDisposedException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, pbeParameters, [], out _));
             Assert.Throws<ObjectDisposedException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>.Empty, pbeParameters, [], out _));
             Assert.Throws<ObjectDisposedException>(() => slhDsa.TryExportPkcs8PrivateKey(tempBuffer, out _));
             Assert.Throws<ObjectDisposedException>(() => slhDsa.TryExportSubjectPublicKeyInfo([], out _));
         }
+
+        public static void AssertImportPublicKey(Action<Func<SlhDsa>> test, SlhDsaAlgorithm algorithm, byte[] publicKey) =>
+            AssertImportPublicKey(test, test, algorithm, publicKey);
+
+        public static void AssertImportPublicKey(Action<Func<SlhDsa>> testDirectCall, Action<Func<SlhDsa>> testEmbeddedCall, SlhDsaAlgorithm algorithm, byte[] publicKey)
+        {
+            testDirectCall(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, publicKey));
+
+            if (publicKey?.Length == 0)
+            {
+                testDirectCall(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, Array.Empty<byte>().AsSpan()));
+                testDirectCall(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, ReadOnlySpan<byte>.Empty));
+            }
+            else
+            {
+                testDirectCall(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, publicKey.AsSpan()));
+            }
+
+            SubjectPublicKeyInfoAsn spki = new SubjectPublicKeyInfoAsn
+            {
+                Algorithm = new AlgorithmIdentifierAsn
+                {
+                    Algorithm = AlgorithmToOid(algorithm) ?? throw new XunitException("Cannot create PKCS#8 private key because algorithm is unknown."),
+                    Parameters = default(ReadOnlyMemory<byte>?),
+                },
+                SubjectPublicKey = publicKey,
+            };
+
+            AssertImportSubjectKeyPublicInfo(import => testEmbeddedCall(() => import(spki.Encode())));
+        }
+
+        internal delegate SlhDsa ImportSubjectKeyPublicInfoCallback(byte[] spki);
+        public static void AssertImportSubjectKeyPublicInfo(Action<ImportSubjectKeyPublicInfoCallback> test) =>
+            AssertImportSubjectKeyPublicInfo(test, test);
```

Comment:
My preference is always that methods in `internal` types are declared `internal`.  I care less in test types.  But I'm confused that you have an `internal` delegate coupled to a `public` method.  That's only legal because the containing type is `internal`, meaning `public` means `internal`.  But the inconsistency is strange.

Reply:
I hadn't thought about it much but I'm realizing it's muscle memory for me to make internal methods in internal classes public and internal types internal. But I think it's more logical to make everything explicitly the most restrictive visibility until otherwise needed.